File: MatrixAddition.java
/*
=====================================================
PRACTICAL: Matrix Addition in Java
=====================================================

AIM:
- To write a Java program that performs addition of two matrices.

OBJECTIVE:
- To understand 2D arrays in Java.
- To implement matrix addition by taking user input.
- To display the resulting matrix.

TOOLS:
- JDK 8+ (javac, java)
- Any IDE or text editor

ALGORITHM:
1) Import Scanner class for input.
2) Ask user for rows and columns.
3) Declare two 2D arrays for the matrices and one for result.
4) Take input for both matrices.
5) Add corresponding elements and store in result matrix.
6) Display the result matrix.

STEPS TO RUN:
- Save as: MatrixAddition.java
- Compile: javac MatrixAddition.java
- Run:     java MatrixAddition

=====================================================
*/

import java.util.Scanner;

public class MatrixAddition {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter number of rows: ");
        int rows = sc.nextInt();
        System.out.print("Enter number of columns: ");
        int cols = sc.nextInt();

        int[][] a = new int[rows][cols];
        int[][] b = new int[rows][cols];
        int[][] sum = new int[rows][cols];

        System.out.println("\nEnter elements of first matrix:");
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                a[i][j] = sc.nextInt();
            }
        }

        System.out.println("\nEnter elements of second matrix:");
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                b[i][j] = sc.nextInt();
            }
        }

        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                sum[i][j] = a[i][j] + b[i][j];
            }
        }

        System.out.println("\nResultant Matrix after Addition:");
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                System.out.print(sum[i][j] + " ");
            }
            System.out.println();
        }

        sc.close();
    }
}

/*
================= SAMPLE OUTPUT =================

Enter number of rows: 2
Enter number of columns: 2

Enter elements of first matrix:
1 2
3 4

Enter elements of second matrix:
5 6
7 8

Resultant Matrix after Addition:
6 8
10 12

=================================================

RESULT:
- Successfully implemented matrix addition in Java.
- Program reads two matrices, adds them, and displays the sum matrix.
=================================================
*/

File: python/matrix addition.py
# ======================================================
# 📘 Practical – 5 : Matrix Addition
# ======================================================

# 🔹 Aim / Objective:
#   To write a Python program to perform addition of 
#   two matrices of same order.

# 🔹 Tools Required:
#   - Python 3.x Interpreter
#   - Text Editor / IDE (IDLE, VS Code, PyCharm etc.)

# 🔹 Algorithm:
#   1. Start program
#   2. Define two matrices A and B of same size
#   3. Create an empty result matrix of same size
#   4. Traverse through rows
#   5. Traverse through columns
#   6. Add corresponding elements A[i][j] + B[i][j]
#   7. Store in result matrix
#   8. Print final matrix
#   9. Stop program

# 🔹 Steps:
#   Step 1: Start program
#   Step 2: Define two matrices A and B
#   Step 3: Create empty result matrix
#   Step 4: Use nested loop to traverse rows and columns
#   Step 5: Add corresponding elements
#   Step 6: Print resulting matrix
#   Step 7: Stop program

# ======================================================
# 🔹 Code
# ======================================================

# Step 1: Define two matrices
A = [[1, 2, 3],
     [4, 5, 6],
     [7, 8, 9]]

B = [[9, 8, 7],
     [6, 5, 4],
     [3, 2, 1]]

# Step 2: Create empty result matrix of same size
result = [[0, 0, 0],
          [0, 0, 0],
          [0, 0, 0]]

# Step 3: Traverse through rows
for i in range(len(A)):
    # Step 4: Traverse through columns
    for j in range(len(A[0])):
        # Step 5: Add corresponding elements
        result[i][j] = A[i][j] + B[i][j]

# Step 6: Print the resulting matrix
print("Matrix Addition Result:")
for row in result:
    print(row)

# ======================================================
# 🔹 Sample Output
# ======================================================
# Matrix Addition Result:
# [10, 10, 10]
# [10, 10, 10]
# [10, 10, 10]
# ======================================================

# 🔹 Result:
# The program successfully performs matrix addition of 
# two matrices of the same order.
# ======================================================
